### PR TITLE
Start porting flatbuffer ops to JSON

### PR DESCRIPTION
**Pull request by ry**
ref #2121 

depends on #2796

next pr contains some examples of ops that have been ported https://github.com/denoland/deno/pull/2802

### CODE REVIEW

**Comment by ry** [comment]
Actually - I've just now started porting flatbuffer ops to JSON. It's going quickly.

I think we should just spend the time to start converting before doing this very radical change.

File: cli/ops/dispatch_json.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  promise_id: Option<u64>,
+}
+
+pub fn dispatch(
+  d: Dispatcher,
+  state: &ThreadSafeState,
+  control: &[u8],
+  zero_copy: Option<PinnedBuf>,
+) -> CoreOp {
+  let async_args: AsyncArgs = serde_json::from_slice(control).unwrap();
```
Comment (bartlomieju) [inline cli/ops/dispatch_json.rs]: A few lines down there's:
```
let result = serde_json::from_slice(control)
    .map_err(ErrBox::from)
    .and_then(move |args| d(state, args, zero_copy));
```
Does it make sense to do `from_slice` operation twice? 

Also to check if op should be sync or async one would have to manually do inside the op:
```
let promise_id = args.promise_id;
let is_sync = promise_id.is_none();
```
Can we some somehow abstract it better?

Comment (ry) [reply cli/ops/dispatch_json.rs]: Improvements welcome.

Comment (ry) [reply cli/ops/dispatch_json.rs]: (Generally I'm trying to not be very prescriptive about the form of the JSON ops, because I'm not sure if there are any odd cases. Once everything is ported over, I think it will make sense to go back and and clean it up.)

File: js/utime.ts [hunk 2]
```diff
@@ -32,7 +17,11 @@ export function utimeSync(
   atime: number | Date,
   mtime: number | Date
 ): void {
-  sendSync(...req(filename, atime, mtime));
+  sendSync(OP_UTIME, {
+    filename,
+    atime: toSecondsFromEpoch(atime),
```
Comment (piscisaureus) [inline js/utime.ts]: I realize this might not be a regression, but to avoid a future addition of a nanosecond-precision API, maybe split this in a [seconds, nanoseconds] tuple.

File: cli/Cargo.toml [hunk 3]
```diff
@@ -37,7 +37,7 @@ regex = "1.2.1"
 remove_dir_all = "0.5.2"
 ring = "~0.14.6"
 rustyline = "5.0.2"
-serde = "1.0.99"
+serde = { version = "1.0.99", features = ["derive"] }
```
Comment (piscisaureus) [inline cli/Cargo.toml]: Preferrably do this as a separate PR, so we can keep the cargo an gn builds in sync.

Comment (piscisaureus) [reply cli/Cargo.toml]: #2808

File: js/utime.ts (hunk 2 above)
Comment (ry) [reply js/utime.ts]: I also think it’s odd that this op takes seconds instead of something more precise but in this PR I’m only trying to change the serialization, not modify the logical messages or behavior of the ops. So I would prefer to do this later.

Comment (ry) [reply js/utime.ts]: I've added a TODO for this.
